### ISSUE TITLE
Fantasque smaller than other fonts at same font size?

### ISSUE THREAD

**Issue by RedBearAK** [belluzj/fantasque-sans#151]
A recent YouTube video about a non-free font called Comic Code prompted me to start comparing it to a number of different free and open source monospace fonts. Fantasque Sans Mono was actually the closest I've come to finding something close to Comic Code. It's very nice. But I noticed something odd about it. 

I don't know much about font design, but I'm really curious whether it's a deliberate design choice to make Fantasque Sans Mono a bit smaller than other similar fonts. I'm using Font Manager from Flathub to compare a number of different fonts and I noticed that Fantasque is just... smaller. I was switching fonts in Visual Studio Code and noticed that I always wanted to zoom in or increase the editor font size whenever I tried Fantasque. The Font Manager app showed that this wasn't just my imagination. It really is relatively smaller at the same "size". 

The screenshot below shows the font compared to four other fonts that share the same purpose of being good monospace "coding" fonts. You can see that there is an unusually large difference in length and height with Fantasque Sans Mono and the other fonts, even though they are all being set to the same "font size" for the comparison. All the other fonts are basically exactly the same length for the same string of characters, not even a single character width of difference. Fantasque has a bit less space between some of the letters, but it's also just physically smaller in general, leading to about a dozen character widths difference in length. 

Is this something done on purpose? Or is it an issue with the settings used while generating the font that might be adjusted to match some sort of font size vs pixel dimensions standard better? Just curious. 

![image](https://user-images.githubusercontent.com/64876997/166892716-5b21a70e-6cc7-4f38-8652-dc916b6ce2b4.png)

**Comment by belluzj** [comment]
It wasn't a conscious decision, that was just due to my inexperience when I made this, if I had done comparisons to other fonts I would have designed the letters with better metrics. I could probably scale all the outlines to match user expectations, but as you found out you can also bump the font size by one or two in the font editor and get something that works for you.

**Comment by RedBearAK** [comment]
@belluzj 

>I could probably scale all the outlines to match user expectations, but as you found out you can also bump the font size by one or two in the font editor and get something that works for you.

Yes, very true. It's not a world-shattering problem, to be sure. And I can't actually find any information about there being an official standard size, even for monospace fonts, other than 72 pt == 1 inch, but that's referring to printed typefaces in the real world, essentially. 

But, as can be seen from the screenshot below, there are a large number of monospace fonts that do seem to be conforming to a remarkably exacting similarity in size-on-screen for a given point size. You can literally draw a straight vertical line down the screen to mark the beginning and end of all the matching font strings. Though there are some notable exceptions, including `Ubuntu Mono` and `Monoid`. 

![monospace_fonts_comparison](https://user-images.githubusercontent.com/64876997/167008976-703bd7bf-55b1-42ad-8f4f-9689014fd271.png)

I feel like the users of FSM would be well served if the font was a lot closer in dimensions to these other monospace fonts. It would allow FSM to more easily be embedded in a paragraph with other fonts without needing to specifically adjust the font size of the embedded string separately from the rest of the paragraph. That's just one example I can think of. 

Thank you for your response.

**Comment by dantezhu** [comment]
@belluzj I have to say that this was a very good decision, because after being two sizes smaller than other fonts, it looks very comfortable when it is the same size as the Chinese font. I remember that inconsolata is also such a font size.

**Comment by sonulohani** [comment]
I dont have problem with the smaller font size. I can increase it anytime.

**Comment by SuggonM** [comment]
The smaller font size issue has been passed down from the original Comic Shanns, then to Comic Mono, and the rest is history.

In fact, I was here to request on making the size consistent with other monospace fonts, or at least releasing a patched variant alongside. For terminals, it might not seem like a big issue. But it causes a myriad of issues elsewhere where any form of GUI is involved and you don't control the font sizes.

For example, if I change the browser's default monospace font to Fantasque, all monospace text elsewhere on the internet will appear smaller and make me squint. If I hit <kbd>Ctrl</kbd> <kbd>+</kbd>, then obviously the whole page will appear larger when that wasn't my intention. Another area is the browser devtools. Sure I can increase the zoom, but the rest of the devtools UI (which is already cramped as-is) will take up more space than they used to. The ridiculous scaling of devtools UI in this screenshot should be enough to visualize the situation.

<img width="2232" height="1568" alt="Image" src="https://github.com/user-attachments/assets/9908e5ac-547a-4700-88b6-f632d63db72d" />

**Comment by RedBearAK** [comment]
@SuggonM 

For anyone who is interested, I have a single variant of the font that's been increased in the "em" size by 10%, so it fits better with other fonts. But it is currently just this one variant of the font that I happen to prefer:

- Spinda fork with no ligatures
- Large Line Height
- No-Loop-K
- The `f` glyph loop is slightly more open
- Size increased by 1.1x
- Patched with Nerd Font script 3.4.0 (11K tech symbols added)

Later on I might put together a collection of the whole set of font variants with these adjustments, which can apparently be scripted pretty easily. 

My repo with the zip is here:
https://github.com/RedBearAK/FantasqueSansMNoLigNerdFont

This has existed for a couple of years and I've been using the font the whole time, but this is the first time I tried adjusting the font's base size, and I updated the Nerd Font patch version. 

The image shows that it's a small adjustment in the size, but the 1.1x makes it fit better with a lot of fonts. Especially in web browsers, where for some reason they all seem to want to set the monospace font to be a few sizes smaller than other font types. I might even look at making a larger variant specifically for that odd legacy behavior in web browsers (16pt/13pt size split that makes monospace fonts smaller). 

<img width="1080" height="640" alt="Image" src="https://github.com/user-attachments/assets/53b78ada-9416-438f-b5af-fb43ea5cd630" />
